ns-gui/src/main/java/de/nocompany/noscrabble/gui/SpielerEingabeController.java

```diff
@@ -0,0 +1,100 @@
+package de.nocompany.noscrabble.gui;
+
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.BorderPane;
+import javafx.stage.Stage;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SpielerEingabeController {
+
+    public Button bSpielStarten;
+    @FXML
+    private Label lVorhandeneSpieler;
+    @FXML
+    private Label lSpieler1;
+    @FXML
+    private Label lSpieler2;
+    @FXML
+    private Label lSpieler3;
+    @FXML
+    private Label lSpieler4;
+    @FXML
+    private TextField tfSpielername;
+    @FXML
+    private Button bSpielerHinzufuegen;
+
+
+    @FXML
+    private List<String> spielerListe = new ArrayList<>();
+
+    private int spielerZaehler = 0;
+    private BorderPane root;
+
+    @FXML
+    void initialize() {
+
+        bSpielerHinzufuegen.setOnAction(e -> doAddSpieler());
+        bSpielStarten.setOnAction(e -> doSpielStarten());
+    }
+
+
+    public void doAddSpieler() {
+        if (spielerZaehler == 0) {
+            lSpieler1.setText(tfSpielername.getText());
+        } else if (spielerZaehler == 1) {
+            lSpieler2.setText(tfSpielername.getText());
+        } else if (spielerZaehler == 2) {
+            lSpieler3.setText(tfSpielername.getText());
+        } else if (spielerZaehler == 3) {
+            lSpieler4.setText(tfSpielername.getText());
+        }
+        spielerZaehler++;
+
+        tfSpielername.setText("");
+        tfSpielername.requestFocus();
+
+        // Verhindere das Hinzufügen von mehr als 4 Spielern
+        if (spielerZaehler >= 4) {
+            bSpielerHinzufuegen.setDisable(true);
+        }
+    }
+
+    public void doSpielStarten() {
+        if (!lSpieler1.getText().isEmpty()) {
```

Comment:
Da die Inhalte ja schon leer sind, kannst Du auch die leere zeichenkette setzen und sparst die die ifs

Reply:
das passst schon so, da im späteren fall man auch Spielernamen löschen kann und dann er in das nächste freie feld ein Namen einträgt